pkg/common/oras/authprovider/azure/azureidentity.go

```diff
@@ -135,8 +135,21 @@ func (d *azureManagedIdentityAuthProvider) Provide(ctx context.Context, artifact
 	serverURL := "https://" + artifactHostName
 
 	// create registry client and exchange AAD token for registry refresh token
-	refreshTokenClient := containerregistry.NewRefreshTokensClient(serverURL)
-	rt, err := refreshTokenClient.GetFromExchange(ctx, "access_token", artifactHostName, d.tenantID, "", d.identityToken.Token)
+	client, err := azcontainerregistry.NewAuthenticationClient(serverURL, nil) // &AuthenticationClientOptions{ClientOptions: options})
+	if err != nil {
+		return provider.AuthConfig{}, re.ErrorCodeAuthDenied.NewError(re.AuthProvider, "", re.AzureWorkloadIdentityLink, err, "failed to create authentication client for container registry by azure managed identity token", re.HideStackTrace)
```

Comment:
nit: we can follow the new pattern to generate a Ratify error: https://github.com/ratify-project/ratify/blob/dev/pkg/controllers/utils/verifier.go#L63

Reply:
@binbin-li  I tried to mimic the pattern that's already being used in this file (and in azureworkloadidentity.go) in multiple places. Should I go ahead and change all of them?